src/AppInstallerCLICore/Workflows/ConfigurationFlow.cpp

```diff
@@ -435,20 +492,40 @@ namespace AppInstaller::CLI::Workflow
                     }
                     else
                     {
-                        AICLI_LOG(Config, Error, << "Configuration unit " << Utility::ConvertToUTF8(unit.UnitName()) << "[" << Utility::ConvertToUTF8(unit.Identifier()) << "] failed with code 0x"
-                            << Logging::SetHRFormat << resultInformation.ResultCode() << " and error message:\n" << Utility::ConvertToUTF8(resultInformation.Description()) << '\n'
-                            << Utility::ConvertToUTF8(resultInformation.Details()));
-                        // TODO: Improve error reporting for failures: use message, known HRs, getting HR system string, etc.
-                        m_context.Reporter.Error() << "  "_liv << Resource::String::ConfigurationUnitFailed << " 0x"_liv << Logging::SetHRFormat << resultInformation.ResultCode() << std::endl;
+                        std::string description = Utility::Trim(Utility::ConvertToUTF8(resultInformation.Description()));
+
+                        AICLI_LOG_LARGE_STRING(Config, Error, << "Configuration unit " << Utility::ConvertToUTF8(unit.UnitName()) << "[" << Utility::ConvertToUTF8(unit.Identifier()) << "] failed with code 0x"
+                            << Logging::SetHRFormat << resultInformation.ResultCode() << " and error message:\n" << description, Utility::ConvertToUTF8(resultInformation.Details()));
+
+                        UnitFailedMessageData messageData = GetUnitFailedData(unit, resultInformation);
+                        auto error = m_context.Reporter.Error();
+                        error << "  "_liv << messageData.Message << std::endl;
+
+                        if (messageData.ShowDescription && !description.empty())
+                        {
+                            constexpr size_t maximumDescriptionLines = 3;

```

Comment:
Might be nice if users could configure this. Maybe just a TODO for now until we know if being able to configure would be useful